IoTPiCar.py

```diff
@@ -41,163 +55,135 @@ def calculate_cycles_from_distance(distance: int):
-#                 print('is detected halting')
-#                 isObjectFound = True
-#                 picar.stop()
-#     picar.stop()
-#     servo.set_angle(0)
-
-def decide_movement(scan_results: list):
-    detection = scan_results[1]
-    forward_domain = detection[3:7]
-    left_domain = detection[0:2]
-    right_domain = detection[8:10]
-
-    print(scan_results[1])
-    isSomethingAhead = sum(forward_domain) >= 1
-    isSomethingLeft = sum(left_domain) >= 1
-    isSomethingRight = sum(right_domain) >= 1
-    if(isSomethingAhead and isSomethingLeft):
-        move(5, movement_type='rotation', direction='right')
-    if(isSomethingAhead and isSomethingRight):
-        move(5, movement_type='rotation', direction='left')
-
-    detection_distances = []
-    safe_forward_cycles = 0
-    print('scan result', scan_results[0][3:8])
-    for scan_result in scan_results[0][3:7]:
-        if(not scan_result['distance'] == -2):
-            detection_distances.append(scan_result['distance'])
+        scan_info.append({'distance': distance, 'angle': next_angle, 'detection': isDetected})
+        detection_list.append(isDetected)       
+        next_angle = get_servo_angle() + servo_delta
+ 
+    #print('One Sweep detection list:', detection_list) 
+    return (scan_info, detection_list)
+
+def decide_movement(sweep_info, Isdetected):
+    isClearAhead = not sum(Isdetected[3:7])
+    isClearLeft = not sum(Isdetected[0:2])
+    isClearRight = not sum(Isdetected[8:10])
+
+    if(isClearAhead):
+        picar.forward(FORWARD_SPEED)
+    else:
+        picar.stop()
+        if(isClearLeft):
+            picar.turn_left(TURN_SPEED)
+        elif(isClearRight):
+            picar.turn_right(TURN_SPEED)
+        else:
+            picar.backward(BACKWARD_SPEED)
+ 
+#     detection_distances = []
+#     safe_forward_cycles = 0
+#     print('scan result', sweep_info[3:8])
+#     print('Isdetected', Isdetected[3:8])
+#     for scan_result in sweep_info[3:7]:
+#         if(not scan_result['distance'] == -2):
+#             detection_distances.append(scan_result['distance'])
+#             
+#     if(not isSomethingAhead):
+#         max_safe_distance = min(detection_distances) -THIRTY_FIVE_CM
+#         print('detection dist', detection_distances)
+#         safe_forward_cycles = calculate_cycles_from_distance(max_safe_distance)
+#         print('safe forward cyucles', safe_forward_cycles)
+#         if(max_safe_distance > 0 and max_safe_distance <= 80):
+#             print('moving forward')
+#             #move(safe_forward_cycles, movement_type='linear', direction='forward')
+
+def create_advanced_mapping(sweep_info):
+    global envoriment_map
+    for entry in sweep_info:
+        if(entry['detection']):
+            #(x,y) with respect to the car
+            scanned_object_x = entry['distance']*math.sin(math.radians(entry['angle']))
+            scanned_object_y = entry['distance']*math.cos(math.radians(entry['angle']))
```

Comment:
is it possible to simplify the math.cos(math.radians into another variable to better show what is happening

Reply:
sure